Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -190,7 +200,19 @@ private void InitializeServiceLocator()
             // If the Spatial Awareness system has been selected for initialization in the Active profile, enable it in the project
             if (ActiveProfile.IsSpatialAwarenessSystemEnabled)
             {
-                if (!RegisterService<IMixedRealitySpatialAwarenessSystem>(ActiveProfile.SpatialAwarenessSystemSystemType) || SpatialAwarenessSystem == null)
+                if (RegisterService<IMixedRealitySpatialAwarenessSystem>(ActiveProfile.SpatialAwarenessSystemSystemType) && SpatialAwarenessSystem != null)
+                {
+                    //if (ActiveProfile.SpatialAwarenessProfile.SpatialObserverDataProviders != null &&
```

Comment:
Along the lines of the earlier comment. If the toolkit component provided a method that systems which use data providers to call at initialization time, we can save a good deal of code here and allow systems / extension services to add/remove data provider requirements without requiring the toolkit component to change.

Reply:
> All services should be registered with the service locator only.

True, though it might be useful / easier to read if the services triggered loading their data providers during initialization.

Reply:
By the time initialization happens all services should already be registered

Reply:
Not a blocker at this time, more of something to consider when data providers are added.